conan/tools/cmake/cmakedeps2/cmakedeps.py

```diff
@@ -243,14 +249,17 @@ def generate(self):
 
         # CMAKE_PROGRAM_PATH | CMAKE_LIBRARY_PATH | CMAKE_INCLUDE_PATH
         cmake_program_path = self._get_cmake_paths([(req, dep) for req, dep in all_reqs if req.direct], "bindirs")
-        cmake_library_path = self._get_cmake_paths(list(host_req.items()) + list(test_req.items()), "libdirs")
-        cmake_include_path = self._get_cmake_paths(list(host_req.items()) + list(test_req.items()), "includedirs")
-
+        cmake_library_path = self._get_cmake_paths(host_test_reqs, "libdirs")
+        cmake_include_path = self._get_cmake_paths(host_test_reqs, "includedirs")
+        cmake_framework_path = []
+        if is_apple_os(self._conanfile):
+            cmake_framework_path = self._get_cmake_paths(host_test_reqs, "frameworkdirs")
```

Comment:
We might want to review this approach at some point, as the ``libdirs/bindirs`` directories are not that necessary if the ``locations`` are specified.

Reply:
You're completely right. I've also removed the `if is_apple_os(self._conanfile):` check as discussed above for `package_frameworks`.